src/routes/publish.js

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// In this router is implemented the publishing operation. After doing various
+// checks the data is uploaded to Google Cloud Storage.
+
+import Router from '@koa/router';
+import Stream from 'stream';
+import util from 'util';
+
+import { getLogger } from '../log';
+import { config } from '../config';
+import { create as gcsStorageCreate } from '../logic/gcs';
+import {
+  HasherPassThrough,
+  LengthCheckerPassThrough,
+  Concatenator,
+} from '../utils/streams';
+
+const MAX_BODY_LENGTH = 32 * 1024 * 1024; // 32MB
+
+export function publishRoutes() {
+  const log = getLogger('routes.publish');
+
+  const router = new Router();
+
+  router.post('/compressed-store', async ctx => {
```

Comment:
One question I haven't decided yet is how we handle the versioning.
Normally it's handled using specific headers, but I find this not very easy to use when accessing APIs from a browser.
Another solution is to attach this to a versioned path, like `/v1`.

For now I decided to not decide, because `/compressed-store` is the backward-compatible version and shouldn't evolve much, hopefully. When I'll implement a WebSocket-based publishing API I'll put it on another endpoint.

What do you think?